modules/runner/src/integrationTest/java/org/apache/ignite/internal/benchmark/AbstractMultiNodeBenchmark.java

```diff
@@ -121,8 +111,77 @@ public final void nodeSetUp() throws IOException {
         }
     }
 
+    /**
+     * Stops the cluster.
+     *
+     * @throws Exception In case of any error.
+     */
     @TearDown
-    public final void nodeTearDown() {
-        IgnitionManager.stop(NODE_NAME);
+    public final void nodeTearDown() throws Exception {
+        List<AutoCloseable> closeables = IntStream.range(0, nodes())
+                .mapToObj(i -> nodeName(BASE_PORT + i))
+                .map(nodeName -> (AutoCloseable) () -> IgnitionManager.stop(nodeName))
+                .collect(toList());
+
+        IgniteUtils.closeAll(closeables);
+    }
+
+    private void startCluster() throws IOException {
+        Path workDir = Files.createTempDirectory("tmpDirPrefix").toFile().toPath();
+
+        String connectNodeAddr = "\"localhost:" + BASE_PORT + '\"';
+
+        List<CompletableFuture<Ignite>> futures = new ArrayList<>();
+
+        @Language("HOCON")
+        String configTemplate = "{\n"
+                + "  \"network\": {\n"
+                + "    \"port\":{},\n"
+                + "    \"nodeFinder\":{\n"
+                + "      \"netClusterNodes\": [ {} ]\n"
+                + "    }\n"
+                + "  },\n"
+                + "  clientConnector: { port:{} },\n"
+                + "  rest.port: {},\n"
+                + "  raft.fsync = " + fsync
+                + "}";
+
+        for (int i = 0; i < nodes(); i++) {
+            int port = BASE_PORT + i;
+            String nodeName = nodeName(port);
+
+            String config = IgniteStringFormatter.format(configTemplate, port, connectNodeAddr,
+                    BASE_CLIENT_PORT + i, BASE_REST_PORT + i);
+
+            futures.add(TestIgnitionManager.start(nodeName, config, workDir.resolve(nodeName)));
+        }
+
+        String metaStorageNodeName = nodeName(BASE_PORT);
+
+        InitParameters initParameters = InitParameters.builder()
+                .destinationNodeName(metaStorageNodeName)
+                .metaStorageNodeNames(List.of(metaStorageNodeName))
+                .clusterName("cluster")
+                .build();
+
+        TestIgnitionManager.init(initParameters);
+
+        for (CompletableFuture<Ignite> future : futures) {
+            assertThat(future, willCompleteSuccessfully());
+
+            if (clusterNode == null) {
+                clusterNode = (IgniteImpl) await(future);
+            } else {
+                await(future);
+            }
+        }
+    }
+
+    private static String nodeName(int port) {
+        return "node_" + port;
+    }
+
+    protected int nodes() {
```

Comment:
It seems that this method, along with the class, should be abstract :thinking:

Reply:
Btw, why not move `clusterSize` parameter to `AbstractMultiNodeBenchmark` and remove this method?

Reply:
it can't be abstract, because `AbstractMultiNodeBenchmark` is state of a benchmark, and class that is state cannot be abstract

> Btw, why not move clusterSize parameter to AbstractMultiNodeBenchmark and remove this method?

I believe that way is more flexible. You could easily introduce new benchmark that will run on a single node or exactly 3 nodes without modifying base class